spring/spring4/spring4-common/src/main/java/org/infinispan/spring/provider/CacheDelegate.java

```diff
@@ -71,13 +76,33 @@ public ValueWrapper get(final Object key) {
 
    @Override
    public <T> T get(Object key, Callable<T> valueLoader) {
-      return (T) nativeCache.computeIfAbsent(key, keyToBeInserted -> {
-         try {
-            return valueLoader.call();
-         } catch (Exception e) {
-            throw ValueRetrievalExceptionResolver.throwValueRetrievalException(key, valueLoader, e);
+      long stamp = synchronousGetLock.readLock();
+      try {
+         T value = (T) nativeCache.get(key);
+         if (value == null) {
+            AtomicBoolean valueObtained = new AtomicBoolean();
```

Comment:
The reason I took `AtomicBoolean` is that I wanted to be 100% sure that thread which is spinning inside the `while` loop will observe its change. Otherwise we can not guarantee that the supplier method will be called exactly once (and this is the contract). Can we guarantee this with `ByRef.Boolean`?